umbraco/Umbraco-CMS
# Media Picker v3

This new media picker supports three new main features, Controlling accepted Media Types, Local Image Crops, and Copy/Paste.

This Media Picker will at some point replace the v2 Media Picker we have today. Most likely v2 will not exist in Umbraco v.9. Tomorrow it will just become an additional choice.

The features of this Media Picker is made clear by looking at the DataType configuration:

![image](https://user-images.githubusercontent.com/6791648/113987077-4193ac00-984e-11eb-8cd2-710049413c18.png)

- Limit picking to certain Media Types.
- Better control of amounts of picked items.
- Set media start-node & ignore user start nodes(same feature as from v2)
- Local Image Croppings


**Test notes:**
- Set local focal point on images.
- Use local focus points in templates.
- Setup image croppings, change, delete, add new ones.
- Pick media and set local crops.
- Use image croppings in templates.
- Limiting accepted media types.
- Set validation limits, min & ma or just min and just max.
- Have pick multiple items turned on but limit the editor to pick maximum one item. (in this way the template still gets an IEnumerable)
- Put Media Item in trash bin and see it presented in the picker.
- Have more items picked than the validation allows
- Test copying media items from one property editor to another.
- Test that you cannot paste none-allowed media types.
- Copy media in media picker v2 to v2.
- Copy media from media picker v2 to v3.
- copy media from v3 to v2.
- copy a collection of media items(done in the property-actions-menu).
- test pasting a collection of media items that contain none-allowed types.(should not be possible)
- Test remove all media, from property-actions
- copy non-image media item from v3 to media picker v2 setup to only accept Image. (should not be possible)
- insert or paste media items at a certain spot in a multiple media picker.
- Read the documentation for ideas on more test cases

**Documentation PR:**
https://github.com/umbraco/UmbracoDocs/pull/3106

**Additional fixes:**
https://github.com/umbraco/Umbraco-CMS/issues/9948
https://github.com/umbraco/Umbraco-CMS/issues/10111
